src/libcore/intrinsics.rs

```diff
@@ -565,8 +565,17 @@ extern "rust-intrinsic" {
     pub fn atomic_umax_rel<T>(dst: *mut T, src: T) -> T;
     pub fn atomic_umax_acqrel<T>(dst: *mut T, src: T) -> T;
     pub fn atomic_umax_relaxed<T>(dst: *mut T, src: T) -> T;
+
+    #[cfg(not(stage0))]
+    pub fn prefetch<T>(data: *const T, rw: i32, locality: i32, cache: i32);
```

Comment:
Yes, you're right. I thought intrinsics APIs should directly map to the underlying llvm API? 
I can add an Enum for cache and locality and make the rw-flag a bool.